src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,46 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
+    private int size;
 
+    public void push(T value) {
+        tail = new Node<>(tail, value);
+        size++;
     }
 
     public T peek() {
-        return null;
+        sizeChecker(size);
+        return tail.item;
     }
 
     public T pop() {
-        return null;
+        sizeChecker(size);
+        T retValue = tail.item;
+        tail = tail.prev;
+        size--;
+        return retValue;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void sizeChecker(int size) {
```

Comment:
```suggestion
    private void checkSize(int size) {
```
sizeChecker it's more likely the name of the class than a method

Reply:
method names should always start with verb